test_pull_requests

```diff
@@ -2323,9 +2341,19 @@ elsif local_merge_pull_id
   @api_client.local_merge_pull_request(local_merge_pull_id, pull_id_repo)
 elsif test_merge_pull_id
   @api_client.test_merge_pull_request(test_merge_pull_id, pull_id_repo, Properties['settings']['merge_test_settings'])
+elsif rate_limit
+  puts "Remaining: #{@api_client.rate_limit_remaining}"
+  puts "Resets in: #{@api_client.rate_limit_reset.to_i - Time.now.to_i}s, at #{Time.at(@api_client.rate_limit_reset.to_i).to_datetime} (#{@api_client.rate_limit_reset})"
 else
   # Process all the pull requests for testing
   $stderr.puts "Processing pull requests..."
-  @api_client.process_pull_requests(merge_pretest_success)
+  $stderr.puts "Rate limit remaining: #{@api_client.rate_limit_remaining}"
+  starting_limit = @api_client.rate_limit_remaining
+  if @api_client.rate_limit_remaining.to_i < 1000
```

Comment:
I still feel like failing gracefully is a better approach, in all scripts, but leaving this at 1000 or 300 should be sufficient -- we don't hit the limit too often, right?